### PR TITLE
fix(challenge): update variable names, assign empty array, update solution code, remove "use strict"

### PR DESCRIPTION
**Pull request by lasjorg**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Challenge: https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals

Right now the same identifier is used in two different scopes and the name isn't that great if you ask me. This updates the inner and outer variable names. If anyone has better names for the variables I'm open for suggestions. But I at least think these are better.

I also assigned an empty array to the inner array instead of `null`, change the instructions to just say the function should return an array, and updated the solution code to just return the mapped array. I also removed the "use strict" as I don't think it is needed, right?

This challenge would likely benefit from more changes but I really just wanted to update the variable names. I was going a bit quick on this so hopefully, I didn't mess anything up.

**Note:** The hints page will need to be updated if/when this is merged.

### CODE REVIEW

**Comment by lasjorg** [comment]
I forgot to update the variable name for the solution code. My bad should be good now, I hope.

**Review by naomi-lgbt — DISMISSED** [review]
LGTM - I've seen a few posts on the forum regarding this, and the confusion often seems to come from the repeated variable names. Thanks for the fix! 🙂

**Comment by RandellDawson** [comment]
This challenge never really made sense.  The user is not using template literal syntax to display anything.  The user is just creating an array of list item strings.  Why not something like the following:

>Use template literal syntax with backticks to create an array of list element strings.  Each list item should be an element of the <code>result</code> object's <code>failure</code> array. Each list element (<code>li</code>) should have the class attribute <code>text-warning</code>. The <code>makeList</code> function should return the array of list item strings.
>
> Use an iterator method (any kind of loop) to get the desired output (shown below).

Then the seed code would be the following (Note that I am using an arrow function, since the user has already learned this syntax):
```js
const result = {
  success: ["max-length", "no-amd", "prefer-arrow-functions"],
  failure: ["no-var", "var-on-top", "linebreak"],
  skipped: ["id-blacklist", "no-dup-keys"]
};
const makeList = (failuresArr) => {
  // Only change code below this line
  const failureItemStrings = [];
  return failureItemStrings;
  // Only change code above this line
}

const failuresList = makeList(result.failure);
```

**Comment by lasjorg** [comment]
@RandellDawson 

I do like some of the changes to the text, but I'm not so sure about this sentence:
> Each list item should be an element of the result object's failure array.

That doesn't really seem to explain where the failure array strings should go in the template literal strings. Although, I am having a bit of a hard time coming up with a succinct way of explaining it without talking about the element text content or the element wrapping like it is now.

I would also go with `failureItems` for the inner variable name (no types on variable names).

**Comment by RandellDawson** [comment]
@lasjorg  Would the following be any better?

> Use template literal syntax with backticks to create an array of list element (<code>li</code>) strings.  Each list element's text should be one of the <code>result</code> object's <code>failure</code> array elements and have a <code>class</code> attribute with the value <code>text-warning</code>. The <code>makeList</code> function should return the array of list item strings.
>
> Use an iterator method (any kind of loop) to get the desired output (shown below).

As far as the inner variable name, I am fine with `failureItems`:
```js
const result = {
  success: ["max-length", "no-amd", "prefer-arrow-functions"],
  failure: ["no-var", "var-on-top", "linebreak"],
  skipped: ["id-blacklist", "no-dup-keys"]
};
const makeList = (failuresArr) => {
  // Only change code below this line
  const failureItems = [];
  return failureItems;
  // Only change code above this line
}

const failuresList = makeList(result.failure);
```

**Comment by lasjorg** [comment]
@RandellDawson Seems pretty good. I have pushed the update.

**Review by RandellDawson — DISMISSED** [review]
LGTM

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ This new way of creating strings gives you more flexibility to create robust str
 
 ## Instructions
 <section id='instructions'>
-Use template literal syntax with backticks to display each entry of the <code>result</code> object's <code>failure</code> array. Each entry should be wrapped inside an <code>li</code> element with the class attribute <code>text-warning</code>, and listed within the <code>resultDisplayArray</code>.
+Use template literal syntax with backticks to create an array of list element (<code>li</code>) strings. Each list element's text should be one of the <code>result</code> object's <code>failure</code> array elements and have a <code>class</code> attribute with the value <code>text-warning</code>. The <code>makeList</code> function should return the array of list item strings.
```
Comment (Manish-Giri) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md:39]: Maybe it's just me, but the wording seemed a bit confusing -

> Each list element's text should be one of the <code>result</code> object's <code>failure</code> array elements

How about 

> Each list element's text should be one of the array elements of the <code>failure</code> property of the <code>result</code> object.

Comment (RandellDawson) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md:39]: @lasjorg What do you think about @Manish-Giri's latest suggestion?

Comment (lasjorg) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-strings-using-template-literals.english.md:39]: @RandellDawson Seem fine either way. The sentence quickly becomes convoluted and hard to follow no matter what. The "of the, of the, of the" seem a little hard to follow as well.

How about "from" the array "on" the object?

> Each list element's text should be one of the array elements from the `failure` property on the `result` object.

**Comment by lasjorg** [comment]
@RandellDawson I have updated the instructions with the changes suggested by @Manish-Giri, but with the two changes to the sentence, I proposed.

**Review by Manish-Giri — APPROVED** [review]
LGTM

**Review by moT01 — APPROVED** [review]
LGTM @lasjorg 🎉 Thanks for contributing to freeCodeCamp 🎉
